### PR TITLE
Add read timeout to HTTP Client

### PR DESCRIPTION
**Pull request by benoist**
Add read timeout with TimeSpan to HTTP Client

### CODE REVIEW

File: spec/std/http/client_spec.cr [hunk 1]
```diff
@@ -24,5 +25,23 @@ module HTTP
...
+      client.read_timeout = 1.second
+      client.get("/")
+
+      expect_raises(IO::Timeout, "read timed out") do
+        client.read_timeout = 0.0001
+        client.get("/")
+      end
+
+      server.close
+    end
```
Comment (jhass) [inline spec/std/http/client_spec.cr]: Feels a bit racey. Maybe make the server sleep for half a second.

Comment (benoist) [inline spec/std/http/client_spec.cr]: Fixed

**Comment by jhass** [comment]
I think this is obvious enough, thank you!

**Comment by asterite** [comment]
Should we also need a write timeout?

**Comment by jhass** [comment]
And an open timeout, I guess so, yeah.

**Comment by benoist** [comment]
When will a write operation time out? It should have a wait phase to be able to trigger the timeout. When it finishes after the predefined timeout period without a wait the scheduler cannot check the timeout and will return the write result instead of raising the timeout exception

**Comment by benoist** [comment]
I've tried locally setting the write_timeout, but I couldn't trigger the exception. Have you got any ideas on how to create a testcase for this?

**Pull request merged**